### PR TITLE
Cover Cart CommandHandlers and QueryHandlers by behat tests

### PR DESCRIPTION
**Pull request by tdavidsonas88**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | increases behat tests coverage for Cart domain feature handlers
| Type?         | improvement
| Category?     | TE
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | #14480
| How to test?  | CI tests should pass

CreateEmptyCustomerCart
UpdateProductQuantityInCart
GetCartInformation - there is no GetCartSummaryHandler; instead is GetCartInformation
RemoveProductFromCart

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/17249)

### CODE REVIEW

**Comment by tdavidsonas88** [comment]
@matks Found one more branch related to Behat on my computer so finished the started handlers also.

**Comment by matks** [comment]
```
--- Failed scenarios:
    Features/Scenario/Order/order_from_bo.feature:132
    Features/Scenario/Address/address.feature:35

339 scenarios (337 passed, 2 failed)
4082 steps (4077 passed, 2 failed, 3 skipped)
3m13.20s (110.40Mb)
```

**Comment by tdavidsonas88** [comment]
I think I have the limited abilities how to control fixtures dependent data. Sometimes our written scenarios change the data and then unexpected failures occur - it's really time consuming when this happens. That's why prefer writing fixtures independent code which also has it's own pros/cons. That's why I apologize for changing order_from_bo.feature to run_last_order_from_bo.feature . Could not find the reasons for failing tests when they run in some particular order and change db data and did not want to mess up core team tests also.

File: tests/Integration/Behaviour/Features/Context/Domain/OrderFeatureContext.php [hunk 1]
```diff
@@ -570,4 +574,31 @@ private function getProductQuantitiesByReference(string $orderReference, string
...
+     */
+    public function associateDefaultCarrierToOrder(string $orderReference)
+    {
+        $orderId = SharedStorage::getStorage()->get($orderReference);
+        /** @var OrderForViewing $orderForViewing */
+        $orderForViewing = $this->getQueryBus()->handle(new GetOrderForViewing($orderId));
+        /** @var OrderCarrierForViewing[] $carriers */
+        $carriers = $orderForViewing->getShipping()->getCarriers();
+        // adding carrier to order if it does not have a valid carrier
+        if (count($carriers) == 0 || (int) $carriers[0]->getCarrierId() == 0) {
```
Comment (PierreRambaud) [inline tests/Integration/Behaviour/Features/Context/Domain/OrderFeatureContext.php]:
```suggestion
        if (count($carriers) === 0 || (int) $carriers[0]->getCarrierId() === 0) {
```
or 
```suggestion
        if (empty($carriers[0]->getCarrierId())) {
```

File: tests/Integration/Behaviour/Features/Context/Domain/AddressFeatureContext.php [hunk 2]
```diff
@@ -81,6 +81,9 @@ public function addNewAddressToCustomerWithFollowingDetails(string $customerRefe
     {
         $testCaseData = $table->getRowsHash();
         $customerId = SharedStorage::getStorage()->get($customerReference);
+        if (is_object($customerId)) {
```
Comment (PierreRambaud) [inline tests/Integration/Behaviour/Features/Context/Domain/AddressFeatureContext.php]: Don't mix the type of `$customerReference`

Comment (tdavidsonas88) [reply tests/Integration/Behaviour/Features/Context/Domain/AddressFeatureContext.php]: not sure how to handle this then one feature step expects object and another id.

Comment (PierreRambaud) [reply tests/Integration/Behaviour/Features/Context/Domain/AddressFeatureContext.php]: You should refactor to only use object or id

**Comment by tdavidsonas88** [comment]
the most time consuming problem with behat scenarios is that some of them are dependent on each other (the state of data in the database). And some of them fail only if the whole behat suite is run. Searching for solution.

**Comment by matks** [comment]
Unfortunately I did not find time to finish this PR. I close it for now and link it to issue

**Pull request closed without merging**